tests/templates/test_subroutines.py

```diff
@@ -634,3 +635,204 @@ def circuit(weight):
+        in the circuit."""
+
+        sqg = 10*len(ph) + 72*len(pphh)
+
+        cnots = 0
+        for i_ph in ph:
+            cnots += 4*(i_ph[1]-i_ph[0])
+
+        for i_pphh in pphh:
+            cnots += 16*(i_pphh[1]-i_pphh[0] + i_pphh[3]-i_pphh[2] + 1)
+        N = 6
+        wires = range(N)
+
+        ref_state = np.array([1, 1, 0, 0, 0, 0])
+
+        with qml.utils.OperationRecorder() as rec:
+            UCCSD(weights, wires, ph=ph, pphh=pphh, init_state=ref_state)
+
+        assert len(rec.queue) == sqg + cnots + 1
+
+        for gate in ref_gates:
+            idx = gate[0]
+
+            exp_gate = gate[1]
+            res_gate = rec.queue[idx]
+            assert isinstance(res_gate, exp_gate)
+
+            exp_wires = gate[2]
+            res_wires = rec.queue[idx]._wires
+            assert res_wires == exp_wires
+
+            exp_weight = gate[3]
+            res_weight = rec.queue[idx].parameters
+            if exp_gate != qml.BasisState:
+                assert res_weight == exp_weight
+            else:
+                assert np.allclose(res_weight, exp_weight)
+
+    @pytest.mark.parametrize(
+        ("weights", "ph", "pphh", "init_state", "msg_match"),
+        [
+            ( np.array([-2.8]), [[0, 2]], [], [1, 1, 0, 0],
+             "'init_state' must be a Numpy array"),
+
+            ( np.array([-2.8]), [[0, 2]], [], (1, 1, 0, 0),
+             "'init_state' must be a Numpy array"),
+
+            ( np.array([-2.8]), [[0, 2]], [], np.array([1.2, 1, 0, 0]),
+             "Elements of 'init_state' must be integers"),
+
+            ( np.array([-2.8]), [], [], np.array([1, 1, 0, 0]),
+             "'ph' and 'pphh' lists can not be both empty"),
+
+            ( np.array([-2.8]), None, [[0, 1, 2, 3]], np.array([1, 1, 0, 0]),
+             "'ph' must be a list"),
+
+            ( np.array([-2.8, 1.6]), [0, [1, 2]], [], np.array([1, 1, 0, 0]),
+             "Each element of 'ph' must be a list"),
+
+            ( np.array([-2.8, 1.6]), [["a", 3], [1, 2]], [], np.array([1, 1, 0, 0]),
+             "Each element of 'ph' must be a list of integers"),
+
+            ( np.array([-2.8, 1.6]), [[1.4, 3], [1, 2]], [], np.array([1, 1, 0, 0]),
+             "Each element of 'ph' must be a list of integers"),
+
+            ( np.array([-2.8]), [[0, 2]], None, np.array([1, 1, 0, 0]),
+             "'pphh' must be a list"),
+
+            ( np.array([-2.8, 1.6]), [], [0, [1, 2, 3, 4]], np.array([1, 1, 0, 0]),
+             "Each element of 'pphh' must be a list"),
+
+            ( np.array([-2.8, 1.6]), [], [[0, 1, "a", 3], [1, 2, 3, 4]], np.array([1, 1, 0, 0]),
+             "Each element of 'pphh' must be a list of integers"),
+
+            ( np.array([-2.8, 1.6]), [], [[0, 1, 1.4, 3], [1, 2, 3, 4]], np.array([1, 1, 0, 0]),
+             "Each element of 'pphh' must be a list of integers"),
+
+            ( np.array([-2.8]), [[0, 2]], [], np.array([1, 1, 0, 0, 0]),
+             "'init_state' must be of shape"),
```

Comment:
Is the shape missing here (and below)?

Reply:
I don't think it is needed it as the messages are still unique for ``ph``, ``pphh``, etc.